### PR TITLE
Adds alert report command

### PR DESCRIPTION
**Pull request by JosephSalisbury**
```
17:34:55 ~/go/src/github.com/giantswarm/oncall-scheduler (alert-report)
$ ./oncall-scheduler report alert --opsgenie-api-key=foobar --slack-token=foobar
```

Screen Shot 2019-06-26 at 17 36 47

### CODE REVIEW

File: cmd/report/alert/runner.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			return microerror.Mask(err)
+		}
+	}
+
+	summary, err := opsgenie.GetAlertSummary()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if err := slack.PostAlertSummary(summary); err != nil {
```
Comment (xh3b4sd) [inline cmd/report/alert/runner.go]:
```suggestion
	err = slack.PostAlertSummary(summary)
	if err != nil {
```

File: pkg/opsgenieclient/client_alert.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+func (c *Client) CountAlerts(query string) (int, error) {
+	type OpsgenieCountResponseData struct {
+		Count int `json:"count"`
+	}
+
+	type OpsgenieCountResponse struct {
+		Data OpsgenieCountResponseData `json:"data"`
+	}
+
+	req, err := http.NewRequest(http.MethodGet, "https://api.opsgenie.com/v2/alerts/count", nil)
```
Comment (xh3b4sd) [inline pkg/opsgenieclient/client_alert.go]: Having a constant that is visible in godoc would be cool.

File: pkg/opsgenieclient/client_alert.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+		alertSummary = append(alertSummary, summaryItem)
+	}
+
+	return alertSummary, nil
+}
+
+// getUnixTime returns the UNIX time in milliseconds,
+// shifted into the past by dayShift days if dayShift != 0.
+func (c *Client) getUnixTime(dayShift int) int64 {
```
Comment (xh3b4sd) [inline pkg/opsgenieclient/client_alert.go]: These deserve unit tests.

File: pkg/slackclient/client_summary.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		attachment := c.buildAttachment(summaryItem)
+		attachments = append(attachments, attachment)
+	}
+
+	if _, _, err := c.client.PostMessage(
+		c.channel,
+		slack.MsgOptionAsUser(true),
+		slack.MsgOptionText("Today's OpsGenie Alert Summary!", false),
+		slack.MsgOptionAttachments(attachments...),
+	); err != nil {
```
Comment (xh3b4sd) [inline pkg/slackclient/client_summary.go]: Please don't mix that if possible. It makes reading code way harder.

**Review by xh3b4sd — CHANGES_REQUESTED** [review]
The PR contains too many changes and should be broken down.

**Comment by JosephSalisbury** [comment]
ptalaaaaaaaaaaaaaaa

**Comment by JosephSalisbury** [comment]
Splittin'

**Pull request closed without merging**